src/Workspaces/VisualBasic/Portable/FindSymbols/VisualBasicReferenceFinder.vb

```diff
@@ -0,0 +1,46 @@
+﻿Imports System.Composition
+Imports System.Threading
+Imports System.Threading.Tasks
+Imports Microsoft.CodeAnalysis.FindSymbols.Finders
+Imports Microsoft.CodeAnalysis.Host.Mef
+
+Namespace Microsoft.CodeAnalysis.FindSymbols
+    <ExportLanguageService(GetType(ILanguageServiceReferenceFinder), LanguageNames.VisualBasic), [Shared]>
+    Friend Class VisualBasicReferenceFinder
+        Implements ILanguageServiceReferenceFinder
+
+        Public Async Function DetermineCascadedSymbolsAsync(namedType As INamedTypeSymbol, project As Project, cancellationToken As CancellationToken) As Task(Of IEnumerable(Of ISymbol)) Implements ILanguageServiceReferenceFinder.DetermineCascadedSymbolsAsync
+            If namedType.Language = LanguageNames.VisualBasic AndAlso project.Language = LanguageNames.VisualBasic Then
+                Dim compilation = Await project.GetCompilationAsync(cancellationToken).ConfigureAwait(False)
+
+                ' If this is a WinForms project, then the VB 'my' feature may have synthesized 
+                ' a property that would return an instance of the main Form type for the project.
+                ' Search for such properties and cascade to them as well.
+                Dim myPropertySymbols = GetMatchingMyPropertySymbols(namedType, DirectCast(compilation, VisualBasicCompilation), cancellationToken).Distinct().ToList()
+                If myPropertySymbols.Count > 0 Then
+                    Return myPropertySymbols
+                End If
+            End If
+
+            Return Nothing
+        End Function
+
+        Private Function GetMatchingMyPropertySymbols(symbol As INamedTypeSymbol, compilation As VisualBasicCompilation, cancellationToken As CancellationToken) As IEnumerable(Of IPropertySymbol)
+            Return GetMatchingMyPropertySymbols(symbol, compilation, compilation.GlobalNamespace, cancellationToken).Concat(
```

Comment:
So, I was using this code as my guide:

```vb
        Protected Overrides Function FindRelatedExplicitlyDeclaredSymbol(symbol As ISymbol, compilation As Compilation) As ISymbol
            ' For example: My.Forms.[|LoginForm|]
            ' LoginForm is a SynthesizedMyGroupCollectionPropertySymbol with no Location. Use the
            ' type of this property, the actual LoginForm type itself, for navigation purposes.

            If symbol.IsKind(SymbolKind.Property) AndAlso symbol.IsImplicitlyDeclared Then
                Dim propertySymbol = DirectCast(symbol, IPropertySymbol)
                If propertySymbol.ContainingType IsNot Nothing AndAlso
                   propertySymbol.ContainingType.Name = "MyForms" AndAlso
                   propertySymbol.ContainingType.ContainingNamespace IsNot Nothing AndAlso
                   propertySymbol.ContainingType.ContainingNamespace.IsMyNamespace(compilation) Then

                    Return propertySymbol.Type
                End If
            End If
```

IsMyNamespace then is written like so:

```vb
        Public Function IsMyNamespace(symbol As ISymbol, compilation As Compilation) As Boolean
            If symbol.Kind <> SymbolKind.Namespace OrElse symbol.Name <> "My" Then
                Return False
            End If

            Dim containingNamespace = symbol.ContainingNamespace

            Return containingNamespace IsNot Nothing AndAlso
                  (containingNamespace.IsGlobalNamespace OrElse Object.Equals(containingNamespace, compilation.RootNamespace))
        End Function
```

But if you think it could only be in the root namespace i'll def change!